tools/keytools/otp/otp-keystore-gen.c

```diff
@@ -0,0 +1,91 @@
+/* otp-keystore-primer.c
+ *
+ * Command line utility to create a OTP image
+ *
+ *
+ * Copyright (C) 2024 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+
+#define OTP_SIZE 4096
```

Comment:
Should this be overridable? Does this vary?

Reply:
It's just an upper limit, but the actual size of the OTP should be enforced by the flashing tool. It's useful in the "primer" case, because it prevent silently overflowing the actual OTP storage size. For other components, this is part of the hal declarations (see h5/h7 and how this value is set in the primer)